### ISSUE TITLE
ISO 8601 with offset not processed correctly

### ISSUE THREAD

**Issue by LukeOwncloud** [fail2ban/fail2ban#1342]
### Environment:

Ubuntu 14.04
Installed via `apt-get install fail2ban`
Version v0.8.11

### The issue / Steps to reproduce

I have a log file to the time is written formatted as ISO 8601 by with offset UTC.
That is: `2016-03-01T10:25:30+00:00` is written (which is correct), but is normally represented as `2016-03-01T11:25:30+01:00`. (`11:25:30` is my local time).

It seems, that the offset is ignored. (If understand right: https://github.com/fail2ban/fail2ban/blob/master/fail2ban/server/datedetector.py#L116) This results fail2ban thinking it is 10:25:30.

#### Observed behavior

This causes fail2ban to ignore these line. Log shows: 

    2016-03-01 10:34:07,971 fail2ban.filter : DEBUG  Ignore line since time 1456821247.0 < 1456824847.97 - 600

#### Any additional information

Surely, I can write my log using ISO 8601 with my local offset ("+01:00"). 

#### Expected behavior

However, I expect fail2ban to handle all formats defined by ISO 8601 gracefully. (Note: There are also formats "+0100" and "+01")

**Comment by yarikoptic** [comment]
Sorry... 0.8.11 is way too old. Seek support for it from your distributor or try a new one

**Comment by yarikoptic** [comment]
And if reproduces with recent one, please reopen

**Comment by LukeOwncloud** [comment]
Okay, old version, did not notice. However, judging from the source code I linked, it's probably still the same behavior.

**Comment by yarikoptic** [comment]
I would appreciate if you check if it is the same empirically...

**Comment by yarikoptic** [comment]
Here you could find most recent release package for your Ubuntu http://neuro.debian.net/debian-devel/pool/main/f/fail2ban/

**Comment by sebres** [comment]
No, it's not reproducible in 0.9, I think since we've fixed it in #721 (prev. #720):

test case (used in `tests/files/logs/mysqld-auth`, test time zone `Berlin`):
All 3 time zones (0 +2 -2) taken into account successfully
```
# failJSON: { "time": "2016-03-01T11:25:30", "match": true , "host": "127.0.0.1", "desc": "mysql 5.6 log format" }
2016-03-01T10:25:30+00:00 5907 [Warning] Access denied for user 'root'@'127.0.0.1' (using password: YES)
# failJSON: { "time": "2016-03-01T09:25:30", "match": true , "host": "127.0.0.1", "desc": "mysql 5.6 log format" }
2016-03-01T10:25:30+02:00 5907 [Warning] Access denied for user 'root'@'127.0.0.1' (using password: YES)
# failJSON: { "time": "2016-03-01T13:25:30", "match": true , "host": "127.0.0.1", "desc": "mysql 5.6 log format" }
2016-03-01T10:25:30-02:00 5907 [Warning] Access denied for user 'root'@'127.0.0.1' (using password: YES)
```
